src/UI/UIView.cpp

```diff
@@ -45,596 +54,905 @@ void UIView::draw_ui_callback()
+          }
+
+
         }
-        else
+
+      }
+      ImGui::SameLine();
+      if (ImGui::Button("Reset Camera")) {
+        m_controller_app.get_trackball_camera().reset_camera();
+      }
+      ImGui::BeginTable("##Camera Point of interest (POI) Editor", 1, ImGuiTableFlags_Borders);
+      ImGui::TableSetupColumn("Name", ImGuiTableColumnFlags_WidthFixed, 120.0f);
+
+
+      for (int row = 0; row < m_controller_app.get_simulation().get_simulation_states().size(); ++row)
+      {
+        ImGui::TableNextRow();
+        ImGui::TableSetColumnIndex(0);
+        bool item_is_selected = (row == m_simulation_state_selection) ? true : false;
+        SimulationState sim_state = m_controller_app.get_simulation().get_simulation_state_at(row);
+        if (ImGui::Selectable(sim_state.poi.label.c_str(), item_is_selected, ImGuiSelectableFlags_AllowDoubleClick))
         {
-          if (m_animated)
+          m_simulation_state_selection = row;
+          m_controller_app.get_trackball_camera().set_current_poi(sim_state.poi);
+          m_clip_min = sim_state.min_clip;
+          m_clip_max = sim_state.max_clip;
+          if (ImGui::IsMouseDoubleClicked(0))
           {
-            /*
-             frame by frame animation
+            m_copy_camera_name = sim_state.poi.label;
+            m_camera_name_window_open = true;
+            m_camera_button_action = BUTTON_ACTION::EDIT;
+          }
+        }
+      }
+      ImGui::EndTable();
 
-            unsigned int active_volume = floor(m_frame);
-              unsigned int active_volume2 = ceil(m_frame);
-              double alpha = m_frame - active_volume;
-              tfn_widget[m_selectedVolume].setMinMax(m_volumes[m_selectedVolume][active_volume]->getMin() * alpha + m_volumes[m_selectedVolume][active_volume2]->getMin() * (1.0 - alpha),
-                m_volumes[m_selectedVolume][active_volume]->getMax() * alpha + m_volumes[m_selectedVolume][active_volume2]->getMax() * (1.0 - alpha));
-              if (active_volume < m_volumes[m_selectedVolume].size() && active_volume2 < m_volumes[m_selectedVolume].size())
-                tfn_widget[m_selectedVolume].setBlendedHistogram(m_volumes[m_selectedVolume][active_volume]->getTransferfunction(0), m_volumes[m_selectedVolume][active_volume2]->getTransferfunction(0), alpha);*/
+      if (m_controller_app.get_simulation().get_simulation_states().size() > 0)
+      {
+        SimulationState& sim_state = m_controller_app.get_simulation().get_simulation_state_at(m_simulation_state_selection);
+        std::string edit_text = "Edit Time Frame " + sim_state.poi.label;
+        ImGui::Text(edit_text.c_str());
+
+        ImGui::Text("Axis aligned clip");
+        glm::vec2 bound = { m_clip_min.x * 100 ,m_clip_max.x * 100 };
+        ImGui::DragFloatRange2("X", &bound.x, &bound.y, 0.1f, 0.0f, 100.0f, "Min: %.1f %%", "Max: %.1f %%");
+        m_clip_min.x = bound.x / 100;
+        m_clip_max.x = bound.y / 100;
+
+        ImGui::SameLine();
```

Comment:
this looks very similar to the main clipping sliders from earlier in the file - any way to abstract and reuse this component?

Reply:
Yeah, you are very right. New level of abstractions need to be written. Let me create an issue with this. I also would have to rework the UI code.

Reply:
https://github.com/brown-ccv/VR-Volumeviewer/issues/46